bundles/product-api-schedule-price-import/src/FondOfKudu/Zed/ProductApiSchedulePriceImport/Business/Model/SchedulePriceProductHandler.php

```diff
@@ -57,21 +64,24 @@ public function __construct(
      */
     public function handleProductAbstract(ProductAbstractTransfer $productAbstractTransfer): ProductAbstractTransfer
     {
-        if (!$this->specialPriceAttributesValidator->validate($productAbstractTransfer->getAttributes())) {
-            return $productAbstractTransfer;
-        }
-
+        $productAttributes = $productAbstractTransfer->getAttributes();
+        $hasRequiredProductAttributes = $this->specialPriceAttributesValidator->hasRequiredProductAttributes($productAttributes);
+        $areProductAttributesValid = $this->specialPriceAttributesValidator->validateProductAttributeValues($productAttributes);
         $priceProductScheduleTransfer = $this->schedulePriceProductAbstractModel
             ->getPriceProductScheduleTransfer($productAbstractTransfer->getIdProductAbstract());
 
-        if ($priceProductScheduleTransfer === null) {
+        if ($priceProductScheduleTransfer === null && $hasRequiredProductAttributes && $areProductAttributesValid) {
```

Comment:
You could simplify the conditions for L73 and L81 if we do an early return if the attributes are either invalid or the required attributes are missing right?

Reply:
It’s not that simple, because even if the attributes aren’t valid for creating a new ScheduledPrice, the existing one might still need to be deleted. The point is, if the SpecialPrice attributes are empty, any existing ScheduledPrice still needs to be deleted. If I exit earlier, I don’t reach that part. Admittedly, I’m not really happy with how this is structured, so I’m open for any kind of improvment :)